### PR TITLE
Implemented map with functionality for the left column.

### PR DESCRIPTION
**Pull request by abijeet-dhillon**
Implemented the map with the following functionality:
- scrollable and moveable map
- map populated with all of the weather stations in our data
- ability to click on markers (fire station icons) and view weather station information in a pop up
- on start up, the map opens with a focus around aspen grove weather station (id = 100), opens the pop up for that station, and displays the station id and name in the html element above the map with its lat., long., and elevation in the html elements below

Testing for the map is found in the map.test.js file in the leaflet folder (will move to the testing folder in later commits).

### CODE REVIEW

File: Site/bc_weather_station_dashboard/website/leaflet/leaflet-src.esm.js
Comment (robertyac) [inline Site/bc_weather_station_dashboard/website/leaflet/leaflet-src.esm.js:1]: This shouldn't be uploaded to the GitHub, add to dependancies .txt file.

**Review by robertyac — CHANGES_REQUESTED** [review]
Im getting some errors when I'm trying to run it on the server. See attached images:

Screenshot 2024-02-29 at 11 12 34 AM
Screenshot 2024-02-29 at 11 12 05 AM

File: Site/bc_weather_station_dashboard/website/leaflet/leaflet-src.esm.js
Comment (carsondrobe) [reply Site/bc_weather_station_dashboard/website/leaflet/leaflet-src.esm.js:1]: run `pip freeze > requirements.txt`

**Comment by robertyac** [comment]
I still think leaflet is being uploaded with this pull request, I don't know if you are working on it still but I'm just going through the PR right now.

**Comment by abijeet-dhillon** [comment]
Just fixed it. I scrapped the idea of getting the data from the csv and instead am pulling the data for the map directly from the database. Also, decided to host leaflet through a CDN to avoid excess files and dependencies. Give it a try, it should be working on the server smoothly now.

**Review by robertyac — APPROVED** [review]
Awesome! Looks great I really like the icons and it's great you got the info from the database instead of the csv. 

I went ahead and deleted the copy you had in static, we still have one in `Site/bc_weather_station_dashboard/website/data` 

Other than that LGTM, for future issues if you're still interested in the map you can maybe see if there is an "easy" way to get the users location and set the default station to the closest one as we discussed with Rohit.

**Review by Davis245 — APPROVED** [review]
Great functionality, well done!

**Pull request merged**